sockeye/inference.py

```diff
@@ -1278,16 +1317,24 @@ def _prune(self,
 
     def _beam_search(self,
                      source: mx.nd.NDArray,
-                     source_length: int) -> Tuple[mx.nd.NDArray, mx.nd.NDArray,
-                                                  mx.nd.NDArray, mx.nd.NDArray, Optional[List[BeamHistory]]]:
+                     source_length: int,
+                     raw_constraint_list: List[Optional[constrained.RawConstraintList]] = None) -> Tuple[mx.nd.NDArray,
```

Comment:
The default argument now is no longer correct (None).
Do we need to make this function use a default anyway? _beam_search is an internal function by Translator, only called with `raw_constraint_list` being a proper list (with items being None).
For correctness we could add an assertion in _beam_search to ensure that the length of `raw_constraint_list` is equal to the number of inputs in the source array.